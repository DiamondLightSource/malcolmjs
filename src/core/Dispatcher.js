/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
