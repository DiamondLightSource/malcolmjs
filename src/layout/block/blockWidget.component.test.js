import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { Toolkit } from 'storm-react-diagrams';
import BlockWidget from './blockWidget.component';
import BlockModel from './block.model';

describe('BlockWidget', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  Toolkit.TESTING = true;

  const icon =
    '<svg height="100" width="100">' +
    '<circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />' +
    '</svg>';

  const block = new BlockModel('block 1', 'block 1 description', 'block1');
  block.addIcon(icon);
  block.addBlockPort({ label: 'in 1', input: true });
  block.addBlockPort({ label: 'out 1', input: false });
  block.addBlockPort({ label: 'out 2', input: false });

  it('block renders correctly', () => {
    const wrapper = shallow(<BlockWidget node={block} />);
    expect(wrapper).toMatchSnapshot();
  });
});
