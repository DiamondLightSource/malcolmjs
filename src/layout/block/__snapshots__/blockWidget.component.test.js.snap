// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockWidget block renders correctly 1`] = `
<WithStyles(Paper)
  className="BlockWidget-block-1"
  elevation={8}
>
  <WithStyles(Typography)
    className="BlockWidget-title-2"
  >
    block 1
  </WithStyles(Typography)>
  <div
    className="BlockWidget-blockContents-3"
    style={
      Object {
        "minHeight": 46,
      }
    }
  >
    <div
      className="BlockWidget-inputPortsContainer-4"
    >
      <BlockPortWidget
        key="2"
        node={
          BlockNodeModel {
            "addBlockPort": [Function],
            "addIcon": [Function],
            "description": "block 1 description",
            "extras": Object {},
            "icon": "<svg height=\\"100\\" width=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"red\\" /></svg>",
            "id": "1",
            "label": "block 1",
            "listeners": Object {},
            "locked": false,
            "ports": Object {
              "2": DefaultPortModel {
                "id": "3",
                "in": true,
                "label": "in 1",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "2",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
              "4": DefaultPortModel {
                "id": "5",
                "in": false,
                "label": "out 1",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "4",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
              "6": DefaultPortModel {
                "id": "7",
                "in": false,
                "label": "out 2",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "6",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
            },
            "selected": false,
            "type": "malcolmjsblock",
            "x": 0,
            "y": 0,
          }
        }
        port={
          DefaultPortModel {
            "id": "3",
            "in": true,
            "label": "in 1",
            "links": Object {},
            "listeners": Object {},
            "locked": false,
            "maximumLinks": undefined,
            "name": "2",
            "parent": BlockNodeModel {
              "addBlockPort": [Function],
              "addIcon": [Function],
              "description": "block 1 description",
              "extras": Object {},
              "icon": "<svg height=\\"100\\" width=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"red\\" /></svg>",
              "id": "1",
              "label": "block 1",
              "listeners": Object {},
              "locked": false,
              "ports": Object {
                "2": [Circular],
                "4": DefaultPortModel {
                  "id": "5",
                  "in": false,
                  "label": "out 1",
                  "links": Object {},
                  "listeners": Object {},
                  "locked": false,
                  "maximumLinks": undefined,
                  "name": "4",
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                },
                "6": DefaultPortModel {
                  "id": "7",
                  "in": false,
                  "label": "out 2",
                  "links": Object {},
                  "listeners": Object {},
                  "locked": false,
                  "maximumLinks": undefined,
                  "name": "6",
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                },
              },
              "selected": false,
              "type": "malcolmjsblock",
              "x": 0,
              "y": 0,
            },
            "selected": false,
            "type": "default",
          }
        }
        portColour="#3f51b5"
      />
    </div>
    <div
      className="BlockWidget-iconContents-7"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg height=\\"100\\" width=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"red\\" /></svg>",
          }
        }
      />
    </div>
    <div
      className="BlockWidget-outputPortsContainer-5"
    >
      <BlockPortWidget
        key="4"
        node={
          BlockNodeModel {
            "addBlockPort": [Function],
            "addIcon": [Function],
            "description": "block 1 description",
            "extras": Object {},
            "icon": "<svg height=\\"100\\" width=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"red\\" /></svg>",
            "id": "1",
            "label": "block 1",
            "listeners": Object {},
            "locked": false,
            "ports": Object {
              "2": DefaultPortModel {
                "id": "3",
                "in": true,
                "label": "in 1",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "2",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
              "4": DefaultPortModel {
                "id": "5",
                "in": false,
                "label": "out 1",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "4",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
              "6": DefaultPortModel {
                "id": "7",
                "in": false,
                "label": "out 2",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "6",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
            },
            "selected": false,
            "type": "malcolmjsblock",
            "x": 0,
            "y": 0,
          }
        }
        port={
          DefaultPortModel {
            "id": "5",
            "in": false,
            "label": "out 1",
            "links": Object {},
            "listeners": Object {},
            "locked": false,
            "maximumLinks": undefined,
            "name": "4",
            "parent": BlockNodeModel {
              "addBlockPort": [Function],
              "addIcon": [Function],
              "description": "block 1 description",
              "extras": Object {},
              "icon": "<svg height=\\"100\\" width=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"red\\" /></svg>",
              "id": "1",
              "label": "block 1",
              "listeners": Object {},
              "locked": false,
              "ports": Object {
                "2": DefaultPortModel {
                  "id": "3",
                  "in": true,
                  "label": "in 1",
                  "links": Object {},
                  "listeners": Object {},
                  "locked": false,
                  "maximumLinks": undefined,
                  "name": "2",
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                },
                "4": [Circular],
                "6": DefaultPortModel {
                  "id": "7",
                  "in": false,
                  "label": "out 2",
                  "links": Object {},
                  "listeners": Object {},
                  "locked": false,
                  "maximumLinks": undefined,
                  "name": "6",
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                },
              },
              "selected": false,
              "type": "malcolmjsblock",
              "x": 0,
              "y": 0,
            },
            "selected": false,
            "type": "default",
          }
        }
        portColour="#3f51b5"
      />
      <BlockPortWidget
        key="6"
        node={
          BlockNodeModel {
            "addBlockPort": [Function],
            "addIcon": [Function],
            "description": "block 1 description",
            "extras": Object {},
            "icon": "<svg height=\\"100\\" width=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"red\\" /></svg>",
            "id": "1",
            "label": "block 1",
            "listeners": Object {},
            "locked": false,
            "ports": Object {
              "2": DefaultPortModel {
                "id": "3",
                "in": true,
                "label": "in 1",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "2",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
              "4": DefaultPortModel {
                "id": "5",
                "in": false,
                "label": "out 1",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "4",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
              "6": DefaultPortModel {
                "id": "7",
                "in": false,
                "label": "out 2",
                "links": Object {},
                "listeners": Object {},
                "locked": false,
                "maximumLinks": undefined,
                "name": "6",
                "parent": [Circular],
                "selected": false,
                "type": "default",
              },
            },
            "selected": false,
            "type": "malcolmjsblock",
            "x": 0,
            "y": 0,
          }
        }
        port={
          DefaultPortModel {
            "id": "7",
            "in": false,
            "label": "out 2",
            "links": Object {},
            "listeners": Object {},
            "locked": false,
            "maximumLinks": undefined,
            "name": "6",
            "parent": BlockNodeModel {
              "addBlockPort": [Function],
              "addIcon": [Function],
              "description": "block 1 description",
              "extras": Object {},
              "icon": "<svg height=\\"100\\" width=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"red\\" /></svg>",
              "id": "1",
              "label": "block 1",
              "listeners": Object {},
              "locked": false,
              "ports": Object {
                "2": DefaultPortModel {
                  "id": "3",
                  "in": true,
                  "label": "in 1",
                  "links": Object {},
                  "listeners": Object {},
                  "locked": false,
                  "maximumLinks": undefined,
                  "name": "2",
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                },
                "4": DefaultPortModel {
                  "id": "5",
                  "in": false,
                  "label": "out 1",
                  "links": Object {},
                  "listeners": Object {},
                  "locked": false,
                  "maximumLinks": undefined,
                  "name": "4",
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                },
                "6": [Circular],
              },
              "selected": false,
              "type": "malcolmjsblock",
              "x": 0,
              "y": 0,
            },
            "selected": false,
            "type": "default",
          }
        }
        portColour="#3f51b5"
      />
    </div>
  </div>
  <WithStyles(Typography)
    className="BlockWidget-description-8"
  >
    block 1 description
  </WithStyles(Typography)>
</WithStyles(Paper)>
`;
